octobot_trading/modes/abstract_trading_mode.py

```diff
@@ -354,3 +556,213 @@ async def edit_order(self, order,
+                    self.exchange_manager.bot_id,
+                    commons_tree.get_exchange_path(
+                        self.exchange_manager.exchange_name,
+                        common_enums.InitializationEventExchangeTopics.CANDLES.value
+                    ),
+                    allow_creation=False
+                ).wait(),
+                timeout
+            )
+            if not self.exchange_manager.is_future:
+                return True
+            await asyncio.wait_for(
+                commons_tree.EventProvider.instance().get_or_create_event(
+                    self.exchange_manager.bot_id,
+                    commons_tree.get_exchange_path(
+                        self.exchange_manager.exchange_name,
+                        common_enums.InitializationEventExchangeTopics.CONTRACTS.value
+                    ),
+                    allow_creation=False
+                ).wait(),
+                timeout
+            )
+            return True
+        except (asyncio.TimeoutError, concurrent.futures.TimeoutError):
+            self.logger.error(f"Initialization took more than {timeout} seconds")
+        return False
+
+    async def save_backtesting_data(self):
+        if not self.are_metadata_saved and self.exchange_manager is not None:
+            run_dbs_identifier = databases.RunDatabasesProvider.instance().get_run_databases_identifier(
+                self.exchange_manager.bot_id
+            )
+            run_data_writer = databases.RunDatabasesProvider.instance().get_run_db(self.exchange_manager.bot_id)
+            await run_data_writer.flush()
+            user_inputs = await commons_configuration.get_user_inputs(run_data_writer)
+
+            # TODO remove when proper run storage strategy
+            await self.save_transactions()
+            await self.save_trades()
+            # END TODO
+
+            if not self.__class__.SAVED_RUN_METADATA_DB_BY_BOT_ID.get(self.exchange_manager.bot_id, False):
+                try:
+                    self.__class__.SAVED_RUN_METADATA_DB_BY_BOT_ID[self.exchange_manager.bot_id] = True
+                    async with databases.DBWriter.database(
+                            run_dbs_identifier.get_backtesting_metadata_identifier(),
+                            with_lock=True) as writer:
+                        await basic_keywords.save_metadata(writer, await self.get_trading_metadata(
+                            user_inputs,
+                            run_dbs_identifier,
+                            is_backtesting=True
+                        ))
+                        self.are_metadata_saved = True
+                except Exception:
+                    self.__class__.SAVED_RUN_METADATA_DB_BY_BOT_ID[self.exchange_manager.bot_id] = False
+                    raise
+
+    async def get_trading_metadata(self, user_inputs=None, run_dbs_identifier=None, is_backtesting=True) -> dict:
```

Comment:
I don't understand why this method should be placed in a trading mode

Reply:
right, it's the same thing as the things marked as `#TODO remove when proper run storage strategy`, we need to discuss where to but it. for now i'm moving the method to be next to the ones we need to move

Reply:
👍